lib/vector-core/proto/event.proto

```diff
@@ -3,6 +3,21 @@ package event;
 
 import "google/protobuf/timestamp.proto";
 
+message EventArray {
+  oneof events {
+    LogArray logs = 1;
+    MetricArray metrics = 2;
+  }
+}
+
+message LogArray {
+  repeated Log logs = 1;
+}
+
+message MetricArray {
+  repeated Metric metrics = 1;
+}
+
```

Comment:
Does this mean we are persisting events as batches in the disk buffer? Any implications for the `vector` source and sink?

Reply:
Yes, we are now persisting events as batches… I think. However, the `vector` source and sink will explicitly be sending individual events over, so it shouldn't change anything there. We could probably get a small boost by changing to sending arrays there too, since it should remove a bit of overhead.

Reply:
Gotcha. I believe the `vector` sink is _already_ batching events together in each request:

https://github.com/vectordotdev/vector/blob/26efeb372cbfcb01a5dd8994aa71a724fde25349/src/sinks/vector/v2/sink.rs#L36-L43

I think we would want to preserve that batching behavior rather than passing along the batches as-is from the topology since we need to create batches based on the user's sink configuration (the `batch` settings).

I might be misunderstanding things here though. I just didn't expect to see any protobuf changes as part of this work.

Reply:
Right, it collects the events (sent individually to the sink via a wrapper at this point) together into a batch, and then serializes them into the protocol as individual events not as an array. I don't believe this will change the wire protocol in any way. The protobuf changes are caused by needing to *buffer* the event arrays.

Regarding batching behavior, I agree that batching behavior should remain unchanged. This has the unfortunate implication that a single array of events may result in more than one output batch, as compared to a single event resulting in zero or one batches, so there will be some batching logic changes when (if?) we convert sinks to accept arrays directly.

Reply:
I may be missing something, but I'm not seeing where these new proto types are actually used anywhere around the disk buffer code. If they are and we're concerned about backwards compatibility, I'd be tempted to avoid changing the behavior of the v1 disk buffers or the proto representation at all. Batching in that context is likely already happening at an IO level, so I'm not sure there's much to gain by encoding it within the proto structure itself.

Reply:
They are used by implication by the buffer `TopologyBuilder` as the input/output of the topology is now `EventArray`, which implements `FixedEncodable`.